docs/code_owners.md

```diff
@@ -0,0 +1,53 @@
+This file is a list of the people responsible for ensuring that patches for a
+particular part of Torch-MLIR are reviewed, either by themself or by someone
+else. They are also the gatekeepers for their part of Torch-MLIR, with the final
+word on what goes in or not.
+
+This file follows the conventions of LLVM's
+[CODE_OWNERS.TXT](https://github.com/llvm/llvm-project/blob/main/llvm/CODE_OWNERS.TXT)
+and Clang's
+[CodeOwners.rst](https://github.com/llvm/llvm-project/blob/main/clang/CodeOwners.rst).
+
+--------------------------------------------------------------------------------
+
+### All parts not covered by anyone else
+
+- Sean Silva (@silvasean)
+
```

Comment:
I think there are still some corners that only I have touched. I'm mostly emeritus at this point but am willing to take responsibility as needed. Seems prudent to have a couple of top level folks.